claireyung:dev-MC_4km_jra_ryf+regionalpanan+isf+draft

claireyung opened the pull request:
<!-- use these prompts for changes to configuration branches, skip them for main branch changes -->
**1. Summary**: New ACCESS-rOM3 panantarctic 4km _**with ice shelves**_ config. 
- Currently set to run from a restart because running from scratch requires a very short timestep, so not suitable for optimisation work. See instructions here https://github.com/claireyung/access-om3-configs/blob/dev-MC_4km_jra_ryf%2Bregionalpanan%2Bisf%2Bdraft/ice_shelf_instructions.md
- Note requires a custom mom6 executable [with changes for ice shelves](https://github.com/ACCESS-NRI/MOM6/issues/29), ~~currently a local version since this https://github.com/ACCESS-NRI/ACCESS-OM3/pull/142#issuecomment-3354313736 (which should be identical to my local version) didn't compile~~ now using [access-om3/pr142-17](https://github.com/ACCESS-NRI/ACCESS-OM3/pull/142#issuecomment-3374712675) 
- Based on https://github.com/ACCESS-NRI/access-om3-configs/pull/713 no ice shelf panan, but it diverged, so there are some upstream commits e.g. changes to authors list that I haven't added yet here
- Also has new input files (see config.yaml)
- Using the [version](https://github.com/claireyung/access-om3-configs/tree/ice_shelf_panan-nancheck3-150925files-notidalmix) that I ran for one year with internal tide dissipation off. 
- Has known problems with [frazil/heat conservation ](https://github.com/claireyung/mom6-panAn-iceshelf-tools/issues/43) and is not well evaluated. For testing/performance use only.
- Expected cost is 20kSU/month, walltime 5:30:00/month

What has changed? Added ice shelf cavities

Why was this done? New capability

**2. Issues Addressed:**
<!-- Add links to github issue(s) this is related to -->
-

**3. Dependencies (e.g. on payu, model or om3-scripts)**

This change requires changes to (note required version where true):
- [ ] payu:
- [ ] access-om3:
- [ ] om3-scripts:
<!-- Describe and link to the related changes to dependencies -->

**4. Ad-hoc Testing**

What ad-hoc testing was done? How are you convinced this change is correct (plots are good)?

**5. CI Testing**
<!-- Has the CI-testing been run? -->
- [ ] `!test repro` has been run

**6. Reproducibility**

Is this reproducible with the previous commit? (If not, why not?)
- [ ] Yes
- [ ] No - `!test repro commit` has been run. <!-- add detail below for why it's answer changing --> 

**7. Documentation**
<!--Does this impact documentation? Has the wiki been updated? Have the `docs/MOM_*` files been updated ?-->

The docs folder has been updated with output from running the model?
- [ ] Yes
- [ ] N/A

A PR has been created for updating the documentation?
- [ ] Yes: <!--link-->
- [ ] N/A

**8. Formatting**
<!-- Are changes to MOM_input in the same order as docs/MOM_parameter_docs.short? -->

Changes to MOM_input have been copied from model output in docs/MOM_parameter_docs.short?
- [ ] Yes
- [ ] N/A

**9. Merge Strategy**
<!-- What is the planned merge strategy (Merge commit, Rebase and merge, or squash) ?
If not squash, link to the related issue in the commit descriptions -->

- [ ] Merge commit
- [ ] Rebase and merge
- [ ] Squash

chrisb13 commented:
Discussed with @helenmacdonald @minghangli-uni @chrisb13 @ezhilsabareesh8 (today).

@minghangli-uni is interested to have a look. He has some optimised compiler flags that could be worth trying. (Chris noted from chat with Claire at lunch that the parallel restarts would only lead to modest improvements, and that CICE/NUOPC timestep is shorter than Claire would like.) @minghangli-uni can have a look after the 25km and 100km optimisation is complete (same pipeline -- @chrisb13 @ezhilsabareesh8 are currently working on this).

chrisb13 commented:
Thanks for the update @claireyung.

> Same issue present in #713

Is that something you're planning on copying across or would you like someone from the ocean team to do it?

claireyung commented:
It's on my to do list for this afternoon, as well as removing Mom overrides in that config! Thanks for checking :)

claireyung commented:
Hi @chrisb13,

[As discussed](https://github.com/ACCESS-NRI/access-om3-configs/issues/880), I've been running an RYF control spin-up. I've paused the spin-up at 5.5 years as we are finding large biases in the Getz ice shelf region that appear to be resolved by moving to IAF forcing, so I'm ready to update this RYF PR with the latest config used to run that short spin-up.

I've squashed a bunch of my runlogs/commits of development made since this original draft PR. Hopefully the size of commits is more manageable now and can be simply cherry-picked on top of the squashed commit you made of everything that came before, https://github.com/chrisb13/access-om3-configs/commit/299e0de3aa861defffac5fdc9b150c68181f9f2c

Some notes: the commit below is from where I have branched off an IAF run, so it would be ideal if this doesn't get squashed to avoid future conflicts (though it seems I already changed the hash by doing the cherrypick onto this branch, so not sure how salvageable that will be):
```
[2025-11-27 15:46:01: Run 35] https://github.com/ACCESS-NRI/access-om3-configs/pull/814/commits/f6a785fc4bb23794dfba3cd4e882c03e02055b7c
```

changes made (non-exhaustive list, apologies for my poor documentation of changes strung between many runlog commits)
- `THICKNESSDIFFUSE` off as mentioned above
- changes to melt parameterisation (`CDRAG_SHELF = 0.0075`, `HMIX_SFC_PROP = 2`, `HMIX_UV_SFC_PROP = 2` so that circumpolar melt rate after ~5 years is tuned to be approximately 1300Gt/yr in the RYF config)
- using a new initial conditions file based on the [Yamazaki 2024](https://zenodo.org/records/13148501) climatology ([discussion](https://github.com/claireyung/mom6-panAn-iceshelf-tools/issues/45#issuecomment-3479393966)) to avoid WOA cold bias in the Amundsen Sea shelf region
- updated the boundary condition file to add axes attributes
- updated the tidal roughness file to be zero below the internal tide critical latitude https://github.com/claireyung/mom6-panAn-iceshelf-tools/issues/42#issuecomment-3388292692
- [frazil turned off under ice shelves](https://github.com/mom-ocean/MOM6/commit/94540bc82713958058bad72617fa844abf1ccf88) ([discussion](https://github.com/claireyung/mom6-panAn-iceshelf-tools/issues/43)), [using a scaled tidal amplitude for friction velocity in melt parameterisation](https://github.com/mom-ocean/MOM6/commit/0c71791cfaf7bed88914a702c143e8df109cac42) ([discussion](https://github.com/claireyung/mom6-panAn-iceshelf-tools/issues/42#issuecomment-3375245677))
- Angus fixed one of the segfault bugs (can now increase `DT_THERM` to 800) - https://github.com/ACCESS-NRI/access-om3-configs/issues/863#issuecomment-3531972353
- [code changes to MOM6 ](https://github.com/mom-ocean/MOM6/compare/main...ACCESS-NRI:MOM6:ice-shelf-dev) from above points, now using prerelease exe https://github.com/ACCESS-NRI/ACCESS-OM3/pull/142#issuecomment-3560954391
- layout changes (using a masktable now to ensure parallel IO factor is reasonable, increase in cores from before, https://github.com/ACCESS-NRI/access-om3-configs/issues/832#issuecomment-3440673348)
- ice shelf instructions updated (due to moderate scale up in cores, running the first month at short timestep should now just fit within the 10hr SR maximum walltime so there is no need for a 5 day + 26 day run), but note that running from scratch still requires a short timestep at the start of the run + a manual change to `input.nml`
- added some diagnostics e.g. `Kd_interface`, `volcello` and `h` that Dougie added capability for (https://github.com/ACCESS-NRI/access-om3-configs/issues/688)

There are plenty of issues remaining in this config that may be an area for future development, I plan to create a list of these in the next few weeks. Also, some of the code changes to MOM6 required for ice shelves should probably be made PRs to ACCESS-NRI/MOM6 rather than just sitting on the ice-shelf-dev branch, and possibly PR'd to the whole MOM6 consortium for some of the code changes. I will also try to organise a list of these.

chrisb13 commented:
Closing this PR as it's being picked up again here: https://github.com/ACCESS-NRI/access-om3-configs/pull/1078
([additional notes](https://github.com/ACCESS-NRI/access-om3-configs/issues/880#issuecomment-3775905898))

dougiesquire commented:
(Reopened temporarily to give me permission to push the updated repro checksums to Claire's branch)
